packages/block-library/src/heading/edit.js

```diff
@@ -61,18 +61,21 @@ function HeadingEdit( {
 				onMerge={ mergeBlocks }
 				onSplit={ ( value, isOriginal ) => {
 					let block;
+					let newAttributes = {};
 
 					if ( isOriginal || value ) {
-						block = createBlock( 'core/heading', {
+						newAttributes = {
 							...attributes,
 							content: value,
-						} );
+						};
+						block = createBlock( 'core/heading', newAttributes );
 					} else {
 						block = createBlock( 'core/paragraph' );
 					}
 
 					if ( isOriginal ) {
 						block.clientId = clientId;
+						block.attributes = newAttributes;
```

Comment:
I don't get this line. The block already has these attributes set in the `if ( isOriginal || value )` code block?

Reply:
Since it's not clear to me, it would be nice to document this code in every split function saying that this is dealing with invisible/internal attributes.

Reply:
Added comments in d293e6c.